src/generic_core/uproxy_core.ts

```diff
@@ -638,138 +671,93 @@ export class uProxyCore implements uproxy_core_api.CoreApi {
     ui.update(uproxy_core_api.Update.CORE_UPDATE_AVAILABLE, details);
   }
 
-  public cloudUpdate = (args :uproxy_core_api.CloudOperationArgs): Promise<void> => {
-    // This is the server name recommended by the blog post.
-    const DROPLET_NAME = 'uproxy-cloud-server';
-
-    // Percentage of cloud install progress devoted to deploying.
-    // The remainder is devoted to the install script.
-    const DEPLOY_PROGRESS = 20;
-
-    if (args.providerName !== 'digitalocean') {
+  public cloudUpdate = (args: uproxy_core_api.CloudOperationArgs): Promise<void> => {
+    if (args.providerName !== CLOUD_PROVIDER_NAME) {
       return Promise.reject(new Error('unsupported cloud provider'));
     }
 
-    this.cloudInterfaces_[args.providerName] = this.cloudInterfaces_[args.providerName] || { installer: undefined, provisioner: undefined};
-    const provisionerName = CLOUD_PROVIDER_MODULE_PREFIX + args.providerName;
-    const provisioner = this.cloudInterfaces_[args.providerName].provisioner || freedom[provisionerName]();
-    const installer = this.cloudInterfaces_[args.providerName].installer || freedom['cloudinstall']();
-    this.cloudInterfaces_[args.providerName] = { installer: installer, provisioner: provisioner};
-
-    const destroyModules = () => {
-      freedom[provisionerName].close(provisioner);
-      freedom['cloudinstall'].close(installer);
-      delete this.cloudInterfaces_[args.providerName];
-    };
-
     switch (args.operation) {
       case uproxy_core_api.CloudOperationType.CLOUD_INSTALL:
         if (!args.region) {
           return Promise.reject(new Error('no region specified for cloud provider'));
         }
+        return this.createCloudServer_(args.region);
+      case uproxy_core_api.CloudOperationType.CLOUD_DESTROY:
+        return this.destroyCloudServer_();
+      case uproxy_core_api.CloudOperationType.CLOUD_REBOOT:
+        return this.rebootCloudServer_();
+      default:
+        return Promise.reject(new Error('cloud operation not supported'));
+    }
+  }
 
-        log.debug('deploying cloud server on %1 in %2', args.providerName, args.region);
-        ui.update(uproxy_core_api.Update.CLOUD_INSTALL_STATUS, 'CLOUD_INSTALL_STATUS_CREATING_SERVER');
-        ui.update(uproxy_core_api.Update.CLOUD_INSTALL_PROGRESS, 0);
+  private createCloudServer_ = (region: string) => {
+    log.debug('creating cloud server in %1', region);
 
-        return provisioner.start(DROPLET_NAME, args.region).then((serverInfo: any) => {
-          log.info('created server on digitalocean: %1', serverInfo);
+    ui.update(uproxy_core_api.Update.CLOUD_INSTALL_STATUS, 'CLOUD_INSTALL_STATUS_CREATING_SERVER');
+    ui.update(uproxy_core_api.Update.CLOUD_INSTALL_PROGRESS, 0);
```

Comment:
As long as this is changing, can we initialize the progress bar at 5 or 10 rather than 0? When the progress bar is at 0, we've observed users confusing it with a horizontal divider, and getting worried when nothing seems to be happening. Initializing it at 5 or 10 could make this clearer. Especially if this would cause an animation to show it filling from 0 to its initial value when it's first shown.